### PR TITLE
Add app-data-validate for metainfo.xml

### PR DESCRIPTION
**Pull request by sabbaka**
Add xml header to metainfo.xml

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -96,12 +96,14 @@ dist-gzip: all cockpit-$(PACKAGE_NAME).spec
 	mv node_modules.release node_modules
 
 srpm: dist-gzip cockpit-$(PACKAGE_NAME).spec
+	appstream-util validate-relax --nonet *.metainfo.xml
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply Makefile]: Sure, and that makes sense too. Thanks!

Triggering tests. Please ignore the forever in-progress cockpit ones, long story.

**Comment by martinpitt** [comment]
Haha:

    Makefile:88: *** missing separator (did you mean TAB instead of 8 spaces?).  Stop.

**Comment by sabbaka** [comment]
Updated

**Comment by martinpitt** [comment]
For $DEITY's sake, the webhook was still wrong. Fixed harder now, and  triggering tests manually again.

**Comment by martinpitt** [comment]
I merged your other PR in the meantime, can you please rebase to current master?

   CONFLICT (content): Merge conflict in cockpit-starter-kit.spec.in

**Comment by sabbaka** [comment]
Updated

**Comment by martinpitt** [comment]
OK, but why did you remove the commit that added the XML header? Rebase pilot error?

**Comment by sabbaka** [comment]
Ooops, I'll fix this tomorrow

**Comment by sabbaka** [comment]
Aaaand, it's done

**Comment by martinpitt** [comment]
Hmm, still fails:
```
appstream-util validate-relax --nonet *.metainfo.xml
/bin/sh: appstream-util: command not found
```
This should be reproducible with `make check`?

**Comment by sabbaka** [comment]
I'm not able to fetch latest images. Even after changing to latest release at the moment ( 186 ):
```
bots/image-customize -v -i cockpit -i `pwd`/cockpit-starter-kit-*.noarch.rpm -s /home/kgliebov/starter-kit/test/vm.install centos-7
 x https://8.43.84.246:443/
 x https://209.132.184.41:8493/
image-download: image-download: couldn't find file anywhere: centos-7-857b2f7320a13423803b50ecb9385ab180622252a2b67e1a6eb18bbacd9ffcd4.qcow2
Traceback (most recent call last):
  File "bots/image-customize", line 128, in <module>
    subprocess.check_call(["image-download", args.base_image])
  File "/usr/lib64/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['image-download', 'centos-7']' returned non-zero exit status 1.
make: *** [Makefile:125: /home/kgliebov/starter-kit/test/images/centos-7] Error 1
```
I fetched image for centos-7 using cockpit and then put it in place of the one it looks for:

```
TEST_AUDIT_NO_SELINUX=1 test/check-application
1..1
# ----------------------------------------------------------------------
# testBasic (__main__.TestApplication)
#
Traceback (most recent call last):
  File "test/common/testlib.py", line 725, in setUp
    self.journal_start = self.machine.journal_cursor()
AttributeError: 'VirtMachine' object has no attribute 'journal_cursor'

not ok 1 testBasic (__main__.TestApplication) # duration: 19s

# 1 TESTS FAILED [20s on kgliebov-usersys-redhat-com]
```

I know it's simple PR, but these are the problems blocking it for now.

**Comment by martinpitt** [comment]
@sabbaka : Perhaps your bots/ directory was out of date? I tried it here, and it works fine. I force-pushed to drop the unnecessary `BuildRequires` (the validation happens at release time, not rpm build time).

**Comment by martinpitt** [comment]
/bin/sh: appstream-util: command not found

This is being called in the tasks container (and will also be in the release container), so this is no good indeed. And installing it on the image is unnecessary. Perhaps it *should*  be done as part of the rpm build.

**Comment by martinpitt** [comment]
Conditionalized the check on presence of appstream-util. With that we at least get in the fix.

**Comment by martinpitt** [comment]
@sabbaka : Yesterday I updated our tasks container to have libappstream-utils, so we can do the validation during RPM build now. I sent PR #102 for this.

**Pull request merged**